### ISSUE TITLE
New Expression BeAnnotatedWith

### ISSUE THREAD

**Issue by pfazzi** [phparkitect/arkitect#229]
### Feature Request

Add `BeAnnotatedWith` Expression.

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | yes/no
| BC Break    | yes/no

#### Summary

Use case:

```php
yield Rule::allClasses()
        ->that(new Implement(App\Shared\Domain\Event\Event::class))
        ->should(new BeAnnotatedWith('@psalm-immutable'))
        ->because('we want to enforce immutability');
```

**Comment by ricfio** [comment]
As class naming I suggest `HaveAnnotationMatching` or `HaveAnnotations` (this last one is my preferred) instead of `BeAnnotatedWith`.

**Comment by AlessandroMinoccheri** [comment]
Closed by #255
